pkgs/applications/networking/cluster/terraform/providers/default.nix

```diff
@@ -1,12 +1,18 @@
 { stdenv, lib, buildGoPackage, fetchFromGitHub }:
 let
-  list = import ./data.nix;
-  toDrv = _: data:
-    buildGoPackage rec {
-      inherit (data) pname version;
-      name = "${pname}-${version}";
-      goPackagePath = "github.com/${data.src.owner}/${data.src.repo}";
-      src = fetchFromGitHub data.src;
-    };
+  importer = path:
+    # vsphere is currently broken
+    if builtins.baseNameOf path == "vsphere" then null
+    else
+      import path { inherit buildGoPackage fetchFromGitHub; };
+
+  importDirs = importer: path:
+    let
+      files = builtins.readDir path;
+      dirs = lib.filterAttrs (k: v: v == "directory") files;
+      importDir = dir: _: { name = builtins.baseNameOf dir; value = importer (path + "/${dir}"); };
+    in
+      lib.mapAttrs' importDir dirs;
```

Comment:
@copumpkin what do you think of this method of importing a bunch of derivations? the `importer` could also be a `callPackage`